File: src/main/scala/com/github/kmizu/yapp/SymbolGenerator.scala
package com.github.kmizu.yapp

class SymbolGenerator(val prefix: String) {
  private var count: Int = 0

  def gensym(): Symbol = {
    val result = Symbol.intern(prefix + count)
    count += 1
    result
  }

  def reset() {
    count = 0
  }
}

File: src/main/java/com/github/kmizu/yapp/SymbolGenerator.java
package com.github.kmizu.yapp;

public class SymbolGenerator {
  private String prefix;
  private int   count;
  
  public SymbolGenerator(String prefix) {
    this.prefix = prefix;
  }
  
  public Symbol gensym() {
    return Symbol.intern(prefix + (count++));
  }
  
  public void reset() {
    count = 0;
  }
}
